### ISSUE TITLE
When querying revisions  $post is not an object in get_the_content()

### ISSUE THREAD

**Issue by Shelob9** [WP-API/WP-API#2836]
I am making GET request to `/wp-json/wp/v2/posts/1/revisions/` Error is: `Trying to get property of non-object in wp-includes/post-template.php on line 298`

That is in get_the_content(), where it uses `$post->post_content` L263 is `$post = get_post();` but there is way that will return a `WP_Post` in that context. 

Here is my xdebug output to show how it got to `get_the_content()` https://pastebin.com/p6jDpqjn

I am using:
WordPress 4.7-alpha-38786
REST API Version 2.0-beta15

**Comment by kadamwhite** [comment]
working on this now

**Comment by kadamwhite** [comment]
Looks like the error is originating in this line within class-wp-rest-revisions-controller:

``` php
$excerpt = apply_filters( 'the_excerpt', apply_filters( 'get_the_excerpt', $excerpt, $post ) );
```

If this is changed to `$excerpt = $excerpt` the bug does not occur and proper JSON is returned.

This sequence of filters is traceable to https://github.com/WP-API/WP-API/commit/fb045337f6d20528337e515c2938c6bc70d79254 and https://github.com/WP-API/WP-API/blob/develop/lib/endpoints/class-wp-rest-posts-controller.php#L1248 we are digging in to why it's the case

**Comment by kadamwhite** [comment]
Should be resolved by #2837
